Jenkinsfile

```diff
@@ -0,0 +1,107 @@
+      steps {
+        withCredentials([
+          string(credentialsId: 'ARTIFACTORY_HOST', variable: 'artifactoryHost'),
+          string(credentialsId: 'GITLAB_HOST', variable: 'gitlabHost')
+          ]) {
+          sh '''
+          if [ $BUILD_TYPE == "releases" ]
+          then
+            bumpversion_resp=$($WORKSPACE/env/bin/bumpversion --allow-dirty --list release)
+            result=$(echo $bumpversion_resp | grep -oh 'new_version=.*')
+          else
+            bumpversion_resp=$($WORKSPACE/env/bin/bumpversion --allow-dirty --no-tag --no-commit --list minor)
+            result=$(echo $bumpversion_resp | grep -oh '^[^ ]*')
+            /usr/bin/git reset --hard HEAD
+          fi
+          timestamp=$(date +%s)
+          bumpversion=$(/usr/bin/cut -d'=' -f2 <<<$result)
+          version="${bumpversion}-${timestamp}"
+          echo "version=${version}" > version.properties
+          imageName="${DOCKER_IMAGE_NAME}:${version}"
+          echo "${imageName}" > image_name.txt
+          '''
+          script {
+            contents = readFile 'image_name.txt'
+            def imageName = contents.trim()
+            def artifactoryPort
+            if ("${BUILD_TYPE}" == "releases") {
+              artifactoryPort = '8446'
+            } else {
+              artifactoryPort = '8445'
+            }
+            def dockerImage = docker.build(imageName)
+            sh "docker tag ${imageName} ${artifactoryHost}:${artifactoryPort}/${ARTIFACTORY_PATH}/${imageName}"
+            sh "docker tag ${imageName} ${gitlabHost}:5001/${GITLAB_PATH}/${imageName}"
+            sh "docker tag ${imageName} ${ECR_PATH}/${imageName}"
+            sh "docker tag ${imageName} usgswma/${imageName}"
+          }
+        }
+      }
+    }
+    stage('Publish Docker images') {
+      steps {
+        withCredentials([
+          string(credentialsId: 'ARTIFACTORY_HOST', variable: 'artifactoryHost'),
+          string(credentialsId: 'GITLAB_HOST', variable: 'gitlabHost')
+          ]) {
+          script {
+            contents = readFile 'image_name.txt'
+            def imageName = contents.trim()
+            env.IMAGE_NAME = imageName
+            if ("${BUILD_TYPE}" == "releases") {
+              artifactoryPort = '8446'
+            } else {
+              artifactoryPort = '8445'
+            }
+            //sh "docker push usgswma/${imageName}"
+            withDockerRegistry([credentialsId: 'ARTIFACTORY_UPLOADER_CREDENTIALS', url: "https://${artifactoryHost}:${artifactoryPort}"]) {
+              sh "docker push ${artifactoryHost}:${artifactoryPort}/${ARTIFACTORY_PATH}/${imageName}"
+            }
+            withDockerRegistry([credentialsId: 'jenkins_ci_access_token', url: 'https://${gitlabHost}:5001']) {
+              sh "docker push ${gitlabHost}:5001/${GITLAB_PATH}/${imageName}"
+            }
+            sh "#!/bin/sh -e\n" + "eval \$(aws --region us-west-2 ecr get-login --no-include-email)"
```

Comment:
Would it be technically possible to roll this in a IAM? Not saying we need to do that now though.

Reply:
Yes, any IAM entity can have the permissions. I wasn't sure how to use that via `docker push` so took the easy route.